admin/controllers/Staff_groups.php

```diff
@@ -162,10 +162,21 @@ public function edit() {
 	}
 
 	private function _saveStaffGroup() {
+                $input_post=$this->input->post();
+                 
+                    $permissions=$this->Permissions_model->getPermissionsByIds();
+                    
+                     
+                    foreach($input_post['permissions'] as $permission_id=>$permission_access){
+                        $input_post['permissions'][$permissions[$permission_id]['name']]= $input_post['permissions'][$permission_id];
+                        //$group_permissions[$permission_info['permission_id']]=$group_permissions[$permission_name];
+                        log_message('info',print_r( $input_post['permissions'],true).'@@@@@@@@@@@ permissions assigned to the staff groups of the user');
```

Comment:
no point logging since alert will be displayed

Also... try removing xss_clean validation rule from line 212 to see if form validation will let you put permissions name in input. Seems like much better way to do it, even though i guess you the go ahead earlier, my bad :(